spec/models/user_spec.rb

```diff
@@ -0,0 +1,52 @@
+require "rails_helper"
+
+def test_wrong_user(params = {})
+  wrong_user = User.create(params)
+  expect(wrong_user.persisted?).to be_falsy
+end
+
+RSpec.describe User, type: :model do
+  let!(:email) { "user@example.com" }
+  let!(:password) { "password" }
+  let!(:first_name) { "louis" }
+  let!(:last_name) { "ramos" }
+  let!(:user) { User.create(email:, password:, first_name:, last_name:) }
+
+  describe "validation" do
+    it "requires an email adress, a password, a first name and a last name" do
+      expect(user.persisted?).to be_truthy
+
+      user.destroy
+      test_wrong_user(email:, password:, first_name:)
+      test_wrong_user(email:, password:, last_name:)
+      test_wrong_user(email:, first_name:, last_name:)
+      test_wrong_user(password:, first_name:, last_name:)
+    end
+
+    it "requires a unique email" do
+      test_wrong_user(email:, password:, first_name:, last_name:)
+    end
+
+    it "requires a valid email address" do
+      user.destroy
+      test_wrong_user(email: "user@example", password:, first_name:, last_name:)
+      test_wrong_user(email: "user.example.com", password:, first_name:, last_name:)
+      test_wrong_user(email: "user@example.c", password:, first_name:, last_name:)
+    end
+  end
```

Comment:
Here I make a separation for the validation part of the model. It's not required but it makes the code easier to read

I have a first check that checks that the User model (it) requires the presence of all the fields to be created
That means that if a user has all the params then the user should be persisted
And if a user is missing a param then user user should not be persisted
`test_wrong_user` is a custom method I made at the beginning of the file to be able to repeat the same code over and over again without having to write the same code multiple times.
It tries to create a user with wrong params and it checks that the user is not persisted because we don't want to persists a user than doesn't have all of its params.

Then I check the email needs to be unique by trying to create a second user with the same email address and make sure it fails

Then I test that the email address needs to be correct by trying to create users with wrong email addresses